cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -586,12 +586,35 @@ where
 					.map_err(|_| Self::Error::InvalidStateRootType)
 			};
 
+			// TODO: abstract the execution proof impl to be reusable in the test.
 			let fraud_proof = if local_trace_idx == 0 {
 				// `initialize_block` execution proof.
 				let pre_state_root = as_h256(parent_header.state_root())?;
 				let post_state_root = as_h256(&execution_receipt.trace[0])?;
 
-				let proof = todo!("Create `initialize_block` proof");
+				let new_header = Block::Header::new(
```

Comment:
It is strange that we need block header in this case at all, especially when it looks like this :thinking:

Reply:
Hm, if this is what header actually looks like during execution then fine, just strange construction :slightly_smiling_face:

Reply:
Yeah, it's not quite straightforward as the actual argument for `initialize_block` is constructed inside the BlockBuilder::new().